src/app/core/nav-menu/nav-menu.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { filter, map, withLatestFrom } from 'rxjs/operators';
+import { Observable, of } from 'rxjs';
+
+import { Route, Subroute } from './models';
+import { appNavRoutes } from './routes';
+import { routerSelectors, State } from '../../root-store';
+
+
+@Injectable()
+export class NavMenuService {
+  constructor(private store: Store<State>) {
+  }
+
+  public getRoutes(): Observable<Route[]> {
+    return of(appNavRoutes);
+  }
+
+  public getSubroutes(): Observable<Subroute[]> {
+    return this.getCurrentRoute().pipe(
+      map(route => route.subroutes)
+    );
+  }
+
+  public getCurrentRoute(): Observable<Route> {
+    return this.getCurrentSubroute().pipe(
+      filter(subroute => !!subroute),
```

Comment:
`filter(Boolean)`

Reply:
If you do so you will lose type below in pipe